Build failure: breakpad: src/common/module.cc:194:52: error: 'find_if' is not a member of 'std'

## Steps To Reproduce

Steps to reproduce the behavior:

1. build *breakpad* (used `nixos-rebuild switch --upgrade`)

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Checking that all programs called by absolute paths in udev rules exist... OK
building '/nix/store/1ql7wvj7szls4zdr1rrsf1zmin2gsxqb-unit-script-cups-pre-start.drv'...
building '/nix/store/hnfgjm8a9mc1iy1nd3f9sgcjyk0hzs9w-X-Restart-Triggers-systemd-udevd.drv'...
building '/nix/store/qg9rpqllqv3jbyvvrv81lbdn9dzwa2qq-unit-cups.service.drv'...
building '/nix/store/bjh5qk2wigfxj1wfmb2ggrwhca0sdvc6-unit-systemd-udevd.service.drv'...
building '/nix/store/8650vjxrr71223wcg9qrx6g70nyyl03l-system-units.drv'...
g++ -DHAVE_CONFIG_H -I. -I./src  -I./src   -g -O2 -c -o src/common/tools_linux_dump_syms_dump_syms-dwarf_line_to_module.o `test -f 'src/common/dwarf_line_to_module.cc' || echo './'`src/common/dwarf_line_to_module.cc
g++ -DHAVE_CONFIG_H -I. -I./src  -I./src   -g -O2 -c -o src/common/tools_linux_dump_syms_dump_syms-dwarf_range_list_handler.o `test -f 'src/common/dwarf_range_list_handler.cc' || echo './'`src/common/dwarf_range_list_handler.cc
g++ -DHAVE_CONFIG_H -I. -I./src  -I./src   -g -O2 -c -o src/common/tools_linux_dump_syms_dump_syms-language.o `test -f 'src/common/language.cc' || echo './'`src/common/language.cc
g++ -DHAVE_CONFIG_H -I. -I./src  -I./src   -g -O2 -c -o src/common/tools_linux_dump_syms_dump_syms-module.o `test -f 'src/common/module.cc' || echo './'`src/common/module.cc
src/common/module.cc: In member function 'bool google_breakpad::Module::AddFunction(Function*)':
src/common/module.cc:194:52: error: 'find_if' is not a member of 'std'; did you mean 'find'?
  194 |     FunctionSet::iterator existing_function = std::find_if(
      |                                                    ^~~~~~~
      |                                                    find
make: *** [Makefile:7971: src/common/tools_linux_dump_syms_dump_syms-module.o] Error 1
error: builder for '/nix/store/r7wahqqw1fkw6p00z8khqw7zq1pd9k29-breakpad-2023.06.01.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/5glnb3vsysi996dli0mpl9fnf9lp1hcc-sentry-native-0.7.16.drv' failed to build
error: 1 dependencies of derivation '/nix/store/nlh2hfvxd6sp6p08s4z7hvvxpr3hdbyf-protonmail-bridge-gui-3.15.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/755wb58my4jg5ylj96h8c77ba5jd1pvs-user-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/nqhcklln7671j56v2i7j8f4ns67vhzj6-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/viq8cwzfhg11vmak3z3xfdwk0033r5ir-nixos-system-oankali-25.05beta729082.634fd4680144.drv' failed to build
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.64, NixOS, 25.05 (Warbler), 25.05beta724962.d70bd19e0a38`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(jzacsh): `"nixos"`
 - channels(root): `"nixos, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`


## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@berberman

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/br/breakpad/package.nix
+++ b/pkgs/by-name/br/breakpad/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchgit,
+  fetchpatch,
   zlib,
 }:
 let
@@ -21,6 +22,14 @@ stdenv.mkDerivation (finalAttrs: {
     rev = "v${finalAttrs.version}";
     hash = "sha256-8AkC/8oX4OWAcV21laJ0AeMRB9G04rFc6UJFy7Wus4A=";
   };
+
+  patches = [
+    (fetchpatch {
+      name = "gcc-14-fixes.patch";
+      url = "https://github.com/google/breakpad/commit/898a997855168c0e6a689072fefba89246271a5d.patch";
+      hash = "sha256-OxodMx7XfKiD9j6b8oFvloslYagSSpQn7BPdpMVOoDY=";
+    })
+  ];
 
   buildInputs = [ zlib ];
 

